src/bin/pgcopydb/pgcmd.c

```diff
@@ -350,8 +350,9 @@ pg_dump_db(PostgresPaths *pgPaths,
 
 	SafeURI safeURI = { 0 };
```

Comment:
Now that the connection strings need to be dynamic, the safeURI needs to be freed by the end of the function. We might need a specialized function that takes care of that (two pointers need to be freed in the structure) and we need to call that function before returning (even when we return false).